### PR TITLE
[Internal][SDK/Executor] Extract common flow execution logic to flow executor

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Extract common flow execution logic to flow executor to facilitate the common use of executor server and sdk flow test.

This pull request introduces changes to improve the execution and testing of flows in the `promptflow` codebase. The most important changes include adding a new function `execute_flow` to `flow_executor.py` for executing a flow, modifying the `flow_test` function in `test_submitter.py` to use the `execute_flow` function.

Main execution and testing changes:

* [`src/promptflow/promptflow/executor/flow_executor.py`](diffhunk://#diff-faa6c81d614b7e41b18a42a93139d961d92afa9aa9dd0b72cb6b7176d7541e69R1069-R1121): Added a new function `execute_flow` responsible for executing a flow, including aggregation nodes, and added an import for the `persist_multimedia_data` function. [[1]](diffhunk://#diff-faa6c81d614b7e41b18a42a93139d961d92afa9aa9dd0b72cb6b7176d7541e69R1069-R1121) [[2]](diffhunk://#diff-faa6c81d614b7e41b18a42a93139d961d92afa9aa9dd0b72cb6b7176d7541e69L33-R37)
* `src/promptflow/promptflow/_sdk/_submitter/test_submitter.py`: Modified the `flow_test` function to use the `execute_flow` function instead of `FlowExecutor` object, and updated import statements accordingly. (F44c67af)
* [`src/promptflow/tests/executor/e2etests/test_executor_happypath.py`](diffhunk://#diff-44d4009e9df8029bb88432b7a843d3a887764cd6a67070afc49557334af3cbaaR5-R7): Added imports and a new test method to improve the testing coverage of the `execute_flow` function. [[1]](diffhunk://#diff-44d4009e9df8029bb88432b7a843d3a887764cd6a67070afc49557334af3cbaaR5-R7) [[2]](diffhunk://#diff-44d4009e9df8029bb88432b7a843d3a887764cd6a67070afc49557334af3cbaaR16-R19) [[3]](diffhunk://#diff-44d4009e9df8029bb88432b7a843d3a887764cd6a67070afc49557334af3cbaaR279-R313)

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 1]
```diff
@@ -1062,3 +1066,36 @@ def wrapper(*args, **kwargs):
         return result
 
     return wrapper
+
+
+def flow_execution(
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:1071]: execute_flow?

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 2]
```diff
@@ -1062,3 +1066,63 @@ def wrapper(*args, **kwargs):
...
+
+
+def execute_flow(
+    flow_file: Path,
+    working_dir: Path,
+    output_dir: Path,
+    connections: dict,
+    inputs: Mapping[str, Any],
+    *,
+    func: Optional[str] = None,
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:1078]: Maybe use **kwargs and bypassed to flow.create?

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 3]
```diff
@@ -1062,3 +1066,63 @@ def wrapper(*args, **kwargs):
...
+    storage: Optional[AbstractRunStorage] = None,
+    raise_ex: bool = True,
+    stream_output: bool = False,
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:1081]: Let's only use stream_output parameter

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 4]
```diff
@@ -1062,3 +1066,63 @@ def wrapper(*args, **kwargs):
...
+def execute_flow(
+    flow_file: Path,
+    working_dir: Path,
+    output_dir: Path,
+    connections: dict,
+    inputs: Mapping[str, Any],
+    *,
+    func: Optional[str] = None,
+    storage: Optional[AbstractRunStorage] = None,
+    raise_ex: bool = True,
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:1080]: raise_ex might be a legacy parameter, is there anywhere to use True?

File: src/promptflow/promptflow/executor/flow_executor.py (hunk 3 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py:1081]: If we don't need to run the aggregation nodes, maybe we can use `exec_line` directly. I think this function is used to run the entire flow.

File: src/promptflow/promptflow/executor/flow_executor.py (hunk 4 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py:1080]: Shouldn't we set this parameter to true for the executor server?

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py]: Updated.

File: src/promptflow/promptflow/executor/flow_executor.py (hunk 3 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py]: Added

File: src/promptflow/promptflow/executor/flow_executor.py (hunk 2 above)
Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py]: Updated.

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 5]
```diff
@@ -1062,3 +1066,53 @@ def wrapper(*args, **kwargs):
...
+    :param kwargs: Other keyword arguments to create flow executor.
+    :type kwargs: Any
+    :return: The line result of executing the flow.
+    :rtype: ~promptflow.executor._result.LineResult
+    """
+    flow_executor = FlowExecutor.create(flow_file, connections, working_dir, **kwargs)
+    flow_executor.enable_streaming_for_llm_flow(lambda: enable_stream_output)
+    with _change_working_dir(working_dir):
+        # execute nodes in the flow except the aggregation nodes
+        line_result = flow_executor.exec_line(inputs, index=0, allow_generator_output=enable_stream_output)
```
Comment (thy09) [inline src/promptflow/promptflow/executor/flow_executor.py:1105]: Add comments about `index=0`, I think we need to remove this in the future.

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py]: Added.

**Review by lalala123123 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**